src/chttpd.erl

```diff
@@ -345,28 +405,33 @@ make_uri(Req, Raw) ->
     {[{<<"url">>,Url}, {<<"headers">>,{Headers}}]}.
 %%% end hack
 
+authenticate_request(Req) ->
+    AuthenticationFuns = [
+        {"cookie", fun chttpd_auth:cookie_authentication_handler/1},
+        {"default", fun chttpd_auth:default_authentication_handler/1},
+        {"local", fun chttpd_auth:local_admin_handler/1} %% should be last
```

Comment:
In addition to [these problems](https://github.com/apache/couchdb-couch/pull/81/files#diff-b34b52654741e1d12cd91c2b36b09968L350) you seems eventually drop [`OAuth`](https://github.com/apache/couchdb-couch/blob/master/src/couch_httpd_oauth.erl#L33-L35) support.